### ISSUE TITLE
Music Cues Being Skipped?

### ISSUE THREAD

**Issue by KILAH4716** [opennox/opennox#636]
https://youtu.be/j8OUZlkWU2o?si=RHIf2iLleDHikx_S&t=640

This video was initially just a demonstration of how much faster the playnox.xyz framerate is compared to Nox.exe. I decided to use OpenNox to demonstrate for Nox.exe's behalf because it had windowed support.

At 10:40 while I was playtesting the difference between the 2 games, I accidentally stumbled on a bug where the same music would play if OpenNox improperly closed from the game to the main menu. This may include hitting escape, especially at a death screen, or when hitting the X to close the window, but doing so doesn't close Nox entirely, and instead brings you back to the main menu.

What the X should do is a different question, but there are no musical issues when I tested this with Nox Reloaded, and Nox Reloaded doesn't support Windowed mode so I couldn't test the "X" button.

**Comment by dennwc** [comment]
Good catch! I usually test without sound, so it's easy to miss something like this.

I don't think we touched the sound in any meaningful way, so there's probably some reset function missing when we exit the actual game to the menu.

**Comment by KILAH4716** [comment]
Connecting this issue to #663, the same sound issue happens under the same conditions (most noticeable at a death screen) if you shut the game down with Alt+F4.

**Comment by KILAH4716** [comment]
Update: I was experimenting with the X, and returned to troubleshoot this issue a little bit more. After encountering this bug, I tried exiting the game normally, but the game would not fade the music out, and would only close if the current song finished playing.

If you attempted to interact with OpenNox at any point before then (after confirming you want to quit), OpenNox will start to Not Respond. If you continue interacting with it, you can cause the game to crash prematurely, or you can let the game close on its own after the song finishes playing.
